airbyte-integrations/connectors/source-pagarme/source_pagarme/streams.py

```diff
@@ -0,0 +1,172 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import datetime
+from abc import ABC
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+
+import requests
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class PagarmeStream(HttpStream, ABC):
+    page = 1
+    page_size_limit = 1000
+    primary_key = "id"
+    url_base = "https://api.pagar.me/1/"
+
+    def __init__(self, api_key, start_date=None, **kwargs):
+        super().__init__(**kwargs)
+        self.api_key = api_key
+        self._start_date = start_date
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        if response.json():
+            self.page = self.page + 1
+            return self.page
+        return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = {"count": self.page_size_limit}
+        if next_page_token:
+            params.update({"page": self.page})
+        return params
+
+    def request_body_json(self, **kwargs) -> Optional[Mapping]:
+        body = {"api_key": self.api_key}
+        return body
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        response = response.json()
+        yield from response
+
+
+class IncrementalPagarmeStream(PagarmeStream, ABC):
+    filter_field = ""
+    state_checkpoint_interval = 1000
+
+    @property
+    def cursor_field(self) -> str:
+        return "date_created"
+
+    def request_body_json(self, stream_state: Mapping[str, Any], **kwargs) -> Optional[Mapping]:
+        start_date = self._string_to_timestampmillis(self._start_date)
+        if start_date:
+            if stream_state.get(self.cursor_field):
+                start_date = max(self._string_to_timestampmillis(stream_state[self.cursor_field]), start_date)
+            if self.filter_field == "start_date":
+                body = {"api_key": self.api_key, self.filter_field: f"{start_date}"}
+            else:
+                body = {"api_key": self.api_key, self.filter_field: f">{start_date}"}
+        else:
+            body = {"api_key": self.api_key}
+        return body
```

Comment:
Does it make sense this code block? the start date is a required parameter and else never will be executed.